activerecord/lib/active_record/inheritance.rb

```diff
@@ -192,11 +192,11 @@ def type_condition(table = arel_table)
       # If this is a StrongParameters hash, and access to inheritance_column is not permitted,
```

Comment:
@sikachu Because of the above, adding `to_h` might not be enough… what do you think?